### ISSUE TITLE
Provide nicer implementation of `publishLocal / skip` for 2.0.x series 

### ISSUE THREAD

**Issue by mdedetrich** [sbt/sbt#7170]
As noted by https://github.com/sbt/sbt/pull/7165#discussion_r1123429285, there is a nicer implementation possible for the solution in that PR however it wasn't possible to cleanly implement due to an already existing deprecated function (see https://github.com/sbt/sbt/blob/51da437f22506cfedf73abfbf5a33e969f64f548/main/src/main/scala/sbt/Defaults.scala#L3582-L3587).

Since its possible to remove that deprecated function in the 2.0.x series, once https://github.com/sbt/sbt/pull/7165 is merged into the 2.0.x branch a nicer implementation can be made

**Comment by mdedetrich** [comment]
If you want just assign this to me (just ping me when that PR gets merged into 2.0.x)

**Comment by SethTisue** [comment]
> just ping me when that PR gets merged into 2.0.x

@eed3si9n that's not how it works in this repo is it?

**Comment by eed3si9n** [comment]
develop branch is intended to be sbt 2.x so all we need to do is git merge. (I'm doing that for 2.x to make it easier to maintain 3 branches)

**Comment by mdedetrich** [comment]
@eed3si9n I wanted to work on this however I noticed from https://github.com/sbt/sbt/pull/6746 that it appears you are working on your own local `eed3si9n:wip/sbt-2.x` branch rather than a `sbt-2.x` branch that sits in https://github.com/sbt.

Should I just wait until there is an equivalent `2.0.x` branch in sbt upstream or should I make a pr against `develop`?

**Comment by eed3si9n** [comment]
I have https://github.com/sbt/sbt/pull/6746 as pull request open, which I think needs to be reviewed by @adpi2 and others?
